lib/chef/resource/windows_package.rb

```diff
@@ -42,7 +42,10 @@ def initialize(name, run_context = nil)
       end
 
       # windows can't take array options yet
-      property :options, String
+      property :options, String, sensitive: true
+
+      # lazy used to set default value of sensitive to true if options is set
+      property :sensitive, [ TrueClass, FalseClass ], default: lazy { |r| r.options ? true : false }
```

Comment:
Neither of these options should really be changed, the `options` string is not generally sensitive -- options like "/SkipRules=RebootRequiredCheck /Action=install" don't require sensitive.  Its only if the user supplies passwords in the resource that it should be marked sensitive, and i think its upon the user to determine that the options are sensitive and to mark it so.  The general case of automatically determining that a password has been supplied to a property and marking it magically sensitive for the user has too many positive and negative edge cases and I don't think we should open that can of worms.

Reply:
Even I thought the same but I couldn't find a way to suppress the sensitive data provided in `options`  and wanted review and suggestions so made it sensitive. Is there a way we can handle such cases?

Reply:
It has to be manual on the part of the user to mark the whole resource as sensitive.